File: js/ControlDigit/.solved/Snils.js
import { getDigitsFromLeastSignificant, sumWithWeights } from './ControlDigit'

const weights = Array.from(new Array(9), (_, index) => index + 1);

export default function controlNumber (number) {
    const digits = getDigitsFromLeastSignificant(number);
    const result = sumWithWeights(digits, weights) % 101;

    return result === 100 ? 0 : result;
}

File: python/ControlDigit/Solved/snils.py
from control_digit import get_digits_from_least_significant, sum_with_weights


def snils(number: int) -> int:
    weights = list(range(1, 10))
    digits = get_digits_from_least_significant(number)
    result = sum_with_weights(digits, weights) % 101
    return 0 if result == 100 else result
